tests/test_service.rb

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/service'
+require 'pp'
+
+include Cisco
+
+# TestService - Minitest for feature Service node Utility class
+class TestService < CiscoTestCase
+  @skip_unless_supported = 'service'
+  ###################
+  # Service tests   #
+  ###################
+
+  def test_clear_status
+    Service.clear_status
+    assert_equal(Service.upgraded?, false)
+  end
+
+  def test_delete
+    Service.delete('nxos.7.0.3.I5.0.231.CSCvb69953.bin')
+  end
+
+  def test_delete_boot
+    Service.delete_boot
+  end
+
+  def test_image_version
+    assert_equal(Service.image_version('nxos.7.0.3.I2.1.bin'), '7.0(3)I2(1)')
+  end
+
+  def test_upgrade
+    sleep 30
+    Service.upgrade('nxos.7.0.3.I2.5.bin')
```

Comment:
Ok, so instead of commenting out these tests which will break our develop branch if these images are not present, let's drive the test via a yaml file.  If the yaml file is present then the test runs with the values from the yaml file.  If not, the test is skipped.  Here is an example.

- https://github.com/cisco/cisco-network-node-utils/blob/4d086b4809af1763e97b8223ec7b7416b356004d/tests/test_aaa_authorization_service.rb#L82
- The `preconfig_tacacs_server_access` method gets called by `setup` in the minitest file which is a magic method that gets run before any test in the file gets called.
- Yaml file example: https://github.com/cisco/cisco-network-node-utils/blob/4d086b4809af1763e97b8223ec7b7416b356004d/tests/tacacs_server.yaml.example